syncstorage/tests/functional/test_storage.py

```diff
@@ -1575,7 +1575,7 @@ def test_batch_with_failing_bsos(self):
         endpoint = collection + '?batch={0}&commit=true'.format(batch)
         resp = self.app.post_json(endpoint, bsos)
         self.assertEqual(len(resp.json['failed']), 1)
-        self.assertEqual(len(resp.json['success']), 1)
+        self.assertEqual(len(resp.json['success']), 2)
```

Comment:
ISTM that it should not be possible for `a` to be accepted into the batch, but then later fail to commit as part of the batch.  Either the *entire* commit failed and you get a non-200 response, or it succeeded and included all the things that were successfully added to it.  So I don't really seem any ambiguity, but I could easily be missing something.

> If Request #1 sends a success and then the batch times out and
> is purged for some reason what does that mean to the client? 

It can't assume that `a` will be committed until it has committed the batch, but it's not obvious how including `a` in the final "success" list makes that any easier to keep track of.

I'm tentatively open to changing it if this makes more sense to the client, but would like to get Mark or Thom's input before committing either way.